neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -57,53 +57,53 @@ protected override void OnReceive(object message)
                     break;
                 case MessageCommand.Consensus:
-                    OnInventoryReceived(msg.GetPayload<ConsensusPayload>());
+                    OnInventoryReceived((ConsensusPayload)msg.Payload);
                     break;
                 case MessageCommand.FilterAdd:
-                    OnFilterAddMessageReceived(msg.GetPayload<FilterAddPayload>());
+                    OnFilterAddMessageReceived((FilterAddPayload)msg.Payload);
                     break;
                 case MessageCommand.FilterClear:
                     OnFilterClearMessageReceived();
                     break;
                 case MessageCommand.FilterLoad:
-                    OnFilterLoadMessageReceived(msg.GetPayload<FilterLoadPayload>());
+                    OnFilterLoadMessageReceived((FilterLoadPayload)msg.Payload);
                     break;
                 case MessageCommand.GetAddr:
                     OnGetAddrMessageReceived();
                     break;
                 case MessageCommand.GetBlocks:
-                    OnGetBlocksMessageReceived(msg.GetPayload<GetBlocksPayload>());
+                    OnGetBlocksMessageReceived((GetBlocksPayload)msg.Payload);
                     break;
                 case MessageCommand.GetData:
-                    OnGetDataMessageReceived(msg.GetPayload<InvPayload>());
+                    OnGetDataMessageReceived((InvPayload)msg.Payload);
                     break;
                 case MessageCommand.GetHeaders:
-                    OnGetHeadersMessageReceived(msg.GetPayload<GetBlocksPayload>());
+                    OnGetHeadersMessageReceived((GetBlocksPayload)msg.Payload);
                     break;
                 case MessageCommand.Headers:
-                    OnHeadersMessageReceived(msg.GetPayload<HeadersPayload>());
+                    OnHeadersMessageReceived((HeadersPayload)msg.Payload);
                     break;
                 case MessageCommand.Inv:
-                    OnInvMessageReceived(msg.GetPayload<InvPayload>());
+                    OnInvMessageReceived((InvPayload)msg.Payload);
                     break;
                 case MessageCommand.Mempool:
                     OnMemPoolMessageReceived();
                     break;
                 case MessageCommand.Ping:
-                    OnPingMessageReceived(msg.GetPayload<PingPayload>());
+                    OnPingMessageReceived((PingPayload)msg.Payload);
                     break;
                 case MessageCommand.Pong:
-                    OnPongMessageReceived(msg.GetPayload<PingPayload>());
+                    OnPongMessageReceived((PingPayload)msg.Payload);
                     break;
                 case MessageCommand.Transaction:
-                    if (msg.Payload.Length <= Transaction.MaxTransactionSize)
-                        OnInventoryReceived(msg.GetTransaction());
+                    if (msg.Payload.Size <= Transaction.MaxTransactionSize)
```

Comment:
This is slow again, we should expose the Size in a property maybe

Reply:
The fix for this is more dirty... `msg.PayloadSize`?

Reply:
`Transaction.Size` is slow, but not too slow. In NEO 3.0, we don't have utxo in transactions, so the size of transactions are easy to be calculated.

Reply:
Is not better to store the value?

Reply:
If we decide to cache the size value, we can cache it in `Transaction`.